cmd/object-handlers.go

```diff
@@ -1433,10 +1433,6 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		srcInfo.metadataOnly = false
 	}
 
-	if srcInfo.metadataOnly {
-		gr.Close() // We are not interested in the reader stream at this point close it.
-	}
-
```

Comment:
@poornas not sure why this would be true - this is expected to be closed if we are only doing metadataOnly operation - we should figure out why there is a PutObject() for metadata update.

Reply:
For versioned objects, even metadata updates creates a new version

Reply:
Then we need to make sure to gr.Close() inside `erasure-server-pool.go` when we decide to not do PutObject() @poornas